spec/models/accounting/quickbooks/transaction_creator_spec.rb

```diff
@@ -6,24 +6,126 @@
-      amount: 12.09,
-      loan_id: 2,
-      memo: 'I am memo',
-      description: 'I am a line item description',
-      qb_bank_account_id: 89,
-      qb_customer_id: 3
-    )
+  context 'when qb customer does exist' do
+    let(:qb_customer_id) { 3 }
+    let(:loan_id) { 2 }
+    let(:amount) { 78.20 }
+    let(:memo) { 'I am a memo' }
+    let(:description) { 'I am a line item description' }
+
+    subject do
+      creator.add_disbursement(
+        amount: amount,
+        loan_id: loan_id,
+        memo: memo,
+        description: description,
+        qb_bank_account_id: 89,
+        qb_customer_id: qb_customer_id
+      )
+    end
+
+    context 'generic_service' do
+      it 'creates JournalEntry with 2 line items' do
+        expect(generic_service).to receive(:create) do |arg|
+          expect(arg.line_items.count).to eq 2
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with 2 line items and customer refs' do
+        expect(generic_service).to receive(:create) do |arg|
+          arg.line_items.each do |item|
+            expect(item.journal_entry_line_detail.entity.type).to eq 'Customer'
+          end
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with 2 line items and proper customer id' do
+        expect(generic_service).to receive(:create) do |arg|
+          arg.line_items.each do |item|
+            expect(item.journal_entry_line_detail.entity.entity_ref.value).to eq qb_customer_id
+          end
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with 2 line items and proper amount' do
+        expect(generic_service).to receive(:create) do |arg|
+          arg.line_items.each do |item|
+            expect(item.amount).to eq amount
+          end
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with 2 line items and proper description' do
+        expect(generic_service).to receive(:create) do |arg|
+          arg.line_items.each do |item|
+            expect(item.description).to eq description
+          end
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with proper memo' do
+        expect(generic_service).to receive(:create) do |arg|
+          expect(arg.private_note).to eq memo
+        end
+
+        subject
+      end
+
+      it 'creates JournalEntry with default date' do
+        expect(generic_service).to receive(:create) do |arg|
+          expect(arg.txn_date).to be_nil
+        end
+
+        subject
+      end
+
+      context 'and date is supplied' do
+        let(:date) { 3.days.ago }
+
+        subject do
+          creator.add_disbursement(
+            amount: amount,
+            loan_id: loan_id,
+            memo: memo,
+            description: description,
+            qb_bank_account_id: 89,
+            qb_customer_id: qb_customer_id,
+            date: date
+          )
+        end
+
+        it 'creates JournalEntry with date' do
```

Comment:
This is the one of two specs for the requested feature, everything else is prep for RM:6281